language/move-vm/types/src/natives/function.rs

```diff
@@ -106,12 +106,12 @@ impl NativeResult {
 
 /// Return the native gas entry in `CostTable` for the given key.
 /// The key is the specific native function index known to `CostTable`.
-pub fn native_gas(
+pub fn native_gas<NativeIndex: Into<u8>>(
     table: &CostTable,
-    key: NativeCostIndex,
+    native_table_idx: NativeIndex,
```

Comment:
Change this to `native_table_idx: impl Into<u8>` to make the function signature a bit cleaner?